### PR TITLE
Adding golang.18 image.

### PR DESCRIPTION
**Pull request by ybettan**
Signed-off-by: Yoni Bettan <yonibettan@gmail.com>

### CODE REVIEW

File: clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    go install github.com/openshift/release/tools/gotest2junit@latest && \
+    go install github.com/openshift/imagebuilder/cmd/imagebuilder@latest && \
+    go install gotest.tools/gotestsum@latest && \
+    mv $GOPATH/bin/* /usr/bin/ && \
+    rm -rf $GOPATH/* $GOPATH/.cache && \
+    mkdir $GOPATH/bin && \
+    ln -s /usr/bin/imagebuilder $GOPATH/bin/imagebuilder && \
+    ln -s /usr/bin/goimports $GOPATH/bin/goimports && \
+    curl --fail --retry 3 -L https://github.com/golang/dep/releases/download/v0.5.4/dep-linux-amd64 > /usr/bin/dep && \
+    chmod +x /usr/bin/dep
```
Comment (qbarrand) [inline clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile:45]: `dep` is obsolete. Let's remove it from the image.
```suggestion
```

File: clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    touch /os-build-image && \
+    git config --system user.name origin-release-container && \
+    git config --system user.email origin-release@redhat.com
+
+RUN \
+    set -x &&  \
+    curl --fail --retry 3 -L https://golang.org/dl/go$VERSION.linux-amd64.tar.gz | tar -C /usr/local -xzf - && \
+    go install golang.org/x/tools/cmd/cover@latest \
+        golang.org/x/tools/cmd/goimports@latest && \
+    go install github.com/tools/godep@latest && \
```
Comment (qbarrand) [inline clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile:34]: Let's remove this.
```suggestion
```

File: clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    GOPATH=/go \
+    GOROOT=/usr/local/go \
+    LOGNAME=deadbeef
+
+ENV PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+
+COPY cbs-paas7-openshift-multiarch-el7-build.repo /etc/yum.repos.d/
+RUN yum install -y epel-release && \
+    rpm -V epel-release && \
+    INSTALL_PKGS="bc bind-utils bsdtar bzr ceph-common createrepo device-mapper device-mapper-persistent-data e2fsprogs ethtool file findutils gcc git glibc-static glib2-devel gpgme gpgme-devel hostname iptables jq krb5-devel libassuan libassuan-devel libseccomp-devel libvirt-devel lsof make mercurial nmap-ncat openssl protobuf-compiler protobuf-devel rsync socat systemd-devel sysvinit-tools tar tito tree util-linux wget which xfsprogs zip goversioninfo" && \
```
Comment (qbarrand) [inline clusters/build-clusters/01_cluster/ci/_origin-release-build/golang-1.18/Dockerfile:21]: Not sure we need all of those...

**Comment by qbarrand** [comment]
/cc @petr-muller
since you were reviewing the [PR for 1.17](https://github.com/openshift/release/pull/23531) 🙂

**Review by petr-muller — APPROVED** [review]

**Comment by bbguimaraes** [comment]
I assumed the outstanding comments were yet to be addressed.

/approve
/hold

Cancel the hold if that is not the case.

**Comment by ybettan** [comment]
/unhold
I am not sure if this is required or not. We can always remove it in the near future.

**Pull request merged**